osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -156,21 +171,39 @@ public void Import(params string[] paths)
             notification.State = ProgressNotificationState.Completed;
         }
 
-        private readonly object importLock = new object();
+        private readonly Lazy<OsuDbContext> importContext;
 
         /// <summary>
         /// Import a beatmap from an <see cref="ArchiveReader"/>.
         /// </summary>
         /// <param name="archiveReader">The beatmap to be imported.</param>
         public BeatmapSetInfo Import(ArchiveReader archiveReader)
         {
-            BeatmapSetInfo set = null;
-
             // let's only allow one concurrent import at a time for now.
-            lock (importLock)
-                connection.RunInTransaction(() => Import(set = importToStorage(archiveReader)));
+            lock (importContext)
+            {
+                var context = importContext.Value;
+
+                context.Database.AutoTransactionsEnabled = false;

```

Comment:
Is this needed? Why is this not needed inside Delete()?

Reply:
to be honest i'm not sure this has much of a benefit; requires further testing. basically it runs `DetectChanges` less than otherwise would be. When I get to addressing transactions via a better pattern it will likely be automatically toggled if there are benefits.

If you'd prefer we can remove it for now. Or add it to deletion.

Reply:
Also why is this not being reset back to true?

Reply:
oh, ignore previous comment, i misread which bool this was. it needs to be set for deletions too.